src/s2/s2loop.cc

```diff
@@ -110,7 +110,14 @@ S2Loop::S2Loop(S2Loop&& b)
           b.unindexed_contains_calls_.exchange(0, std::memory_order_relaxed)),
       bound_(std::move(b.bound_)),
       subregion_bound_(std::move(b.subregion_bound_)),
-      index_(std::move(b.index_)) {}
+      index_(std::move(b.index_)) {
+  // `index_` has a pointer to an S2Loop::Shape which points to S2Loop.
+  // But, we've moved to a new address, so get the Shape back out of the index
+  // and update it to point to our new location.
+  if (index_.begin() != index_.end()) {
```

Comment:
You can just do:

```
for (S2Shape* shape : index_) { ...} 
```

rather than checking the iterators directly.

Reply:
Yes, but I just copied it from S2Polygon.

I think for your suggestion better to separate PR with same change in S2Polygon

Reply:
Yup you're probably right =D